packages/compass-data-modeling/src/components/new-diagram/select-collections-step.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  Checkbox,
+  css,
+  FormFieldContainer,
+  SearchInput,
+  SelectList,
+  spacing,
+  SpinLoaderWithLabel,
+  WarningSummary,
+} from '@mongodb-js/compass-components';
+import { usePreference } from 'compass-preferences-model/provider';
+import React from 'react';
+import { useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import {
+  selectCollections,
+  toggleInferRelationships,
+} from '../../store/generate-diagram-wizard';
+import type { DataModelingState } from '../../store/reducer';
+
+const loadingStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[1800],
+  marginBottom: spacing[1800],
+});
+
+const errorStyles = css({
+  marginTop: spacing[600],
+  marginBottom: spacing[600],
+});
+
+const selectListStyles = css({
+  maxHeight: 200,
+  overflow: 'scroll',
+});
+
+type SelectCollectionsStepProps = {
+  collections: string[];
+  selectedCollections: string[];
+  automaticallyInferRelationships: boolean;
+  isFetchingCollections: boolean;
+  error?: Error;
+  onCollectionsSelect: (colls: string[]) => void;
+  onAutomaticallyInferRelationshipsToggle: (newVal: boolean) => void;
+};
+
+const SelectCollectionsStep: React.FunctionComponent<
+  SelectCollectionsStepProps
+> = ({
+  automaticallyInferRelationships,
+  collections,
+  selectedCollections,
+  isFetchingCollections,
+  error,
+  onCollectionsSelect,
+  onAutomaticallyInferRelationshipsToggle,
+}) => {
+  const showAutoInferOption = usePreference(
+    'enableAutomaticRelationshipInference'
+  );
+  const [searchTerm, setSearchTerm] = useState('');
+  const filteredCollections = useMemo(() => {
+    try {
+      const regex = new RegExp(searchTerm, 'i');
+      return collections.filter((x) => regex.test(x));
+    } catch {
+      return collections;
+    }
+  }, [collections, searchTerm]);
+
+  if (isFetchingCollections) {
+    return (
+      <div className={loadingStyles}>
+        <SpinLoaderWithLabel progressText="">
+          Fetching collections ...
+        </SpinLoaderWithLabel>
+      </div>
+    );
+  }
+
+  if (error) {
+    return (
+      <div className={errorStyles}>
+        <WarningSummary warnings={[error.message]} />
+      </div>
+    );
+  }
```

Comment:
All of the code in this file was already existent in `new-diagram-form` except for these two views.